### PR TITLE
Work on Bidding

### PR DESCRIPTION
**Pull request by ashfurrow**
Work on bidding, not done yet but the infrastructure is there.

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            1 Error
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG changes made</td>
    </tr>
    
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            3 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">PR is classed as Work in Progress</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |242.3ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-bidding/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
170.7ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-bidding/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
58.1ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-bidding/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
Okay, I've got @orta's changes merged in, patched up my tests, and got this "working." Why scare quotes? Well Causality doesn't like our bids. I keep getting this response event: `ConnectionUnauthorized`.

I've poked around Causality, and it looks like it doesn't have the authorization it needs, the second element in the tuple caught [here](https://github.com/artsy/causality/blob/03be62fb3a614dd5a531a93f544c39261bded275/src/main/scala/net/artsy/auction/routes/WebSocketApi.scala#L160) is `None`. I've compared our sent even JSON with Prediction's but I don't see a difference. Needs more investigation, but this PR is good for review now.

File: Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift [hunk 1]
```diff
@@ -68,7 +68,7 @@ class AuctionLotMetadataStackScrollView: ORStackScrollView {
             if let blurb = lot.lotArtistBlurb {
                 artistBlurb.text = blurb
             } else {
-                artistMetadata.forEach { $0.removeFromSuperview() }
+                artistMetadata.forEach { stack.removeSubview($0) }
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift]: Without using `ORStackView`'s `removeSubview`, views have constraints added to them later with no common superview, causing a crash.

Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift]: nice, good spot!

File: Podfile [hunk 2]
```diff
@@ -107,6 +107,7 @@ target 'Artsy' do
   pod 'Then'
   pod 'Interstellar/Core', git: 'https://github.com/ashfurrow/Interstellar.git', branch: 'observable-merge'
   pod 'Starscream'
+  pod 'SwiftyJSON'
```
Comment (orta) [inline Podfile]: Think this'll get used enough to warrant another library?

Comment (ashfurrow) [inline Podfile]: I do; I've only switched away from manual-JSON-parsing-in-Swift in [these two functions](https://github.com/artsy/eigen/pull/1497/files#diff-6d73ebd58fdd2d00c32813f60608fbd1L45) and it resulted in much much shorter and more readable code. We've got a lot of causality-specific JSON  parsing left to do, and I'd like to change what we've already done as we modify it.

Comment (orta) [inline Podfile]: 👍

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionStateManager.swift [hunk 3]
```diff
@@ -47,14 +51,25 @@ class LiveAuctionStateManager: NSObject {
         socketCommunicator.currentLotUpdate.subscribe { [weak self] update in
             self?.stateReconciler.processCurrentLotUpdate(update)
         }
+
+        socketCommunicator.postEventResponses.subscribe { [weak self] response in
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionStateManager.swift]: oh, interesting, didn't know this was a built in concept

Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionStateManager.swift]: ah - it's not, disregard

**Comment by orta** [comment]
Yep 👍

**Pull request merged**